examples/Quantinuum_advanced_compilation.ipynb

```diff
@@ -0,0 +1,607 @@
+    "\n",
+    "The workflow involves specifying the use of the desired native gate both in the `tket` compiler options as well as the hardware options when submitting circuits with `process_circuit`.\n",
+    "\n",
+    "<div style=\"text-align: center;\">\n",
+    "         <img src=\"figures/native-gates-workflow.png\" width=\"300\" />\n",
+    "</div>\n",
+    "\n",
+    "First we import the functions we need in `tket`.   "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c5cd7eda-ed96-4b84-896f-103349601b43",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "from pytket import Circuit, OpType\n",
+    "from pytket.circuit.display import render_circuit_jupyter\n",
+    "\n",
+    "from pytket.extensions.quantinuum import QuantinuumBackend\n",
+    "from pytket.extensions.quantinuum.backends.quantinuum import (\n",
+    "    QuantinuumBackendCompilationConfig,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fc930451",
+   "metadata": {},
+   "source": [
+    "### Decomposing a random $SU(4)$ unitary into a circuit primitive\n",
+    "\n",
+    "To set up the Quantum Volume test, we start by building up the repeated circuit elements. The function in the code cell below defines a `tket` Circuit Box with [`pytket.circuit.CircBox`](https://tket.quantinuum.com/api-docs/circuit.html#pytket.circuit.CircBox). Circuit Boxes are useful for composing larger circuits from smaller subcircuits that utilize the same set of gates.\n",
+    "\n",
+    "The Circuit box below contains a blueprint for the decomposition of a random generalized $SU(4)$ unitary distributed with the Haar Measure into a circuit primitive over 2-qubits. The implementation is based on [arxiv.0609050](http://arxiv.org/abs/math-ph/0609050). The random $SU(4)$ unitary is generated using [`scipy.stats.unitary_group`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.unitary_group.html#scipy-stats-unitary-group)."
```

Comment:
Are the generated matrices SU(4) or U(4)?

Reply:
They should be SU(4)

Reply:
```python
from scipy.linalg import det
r = unitary_group.rvs(dim=4)
det(r)
```
I've done a quick check and it seems that the determinant is not always 1.

Reply:
thanks

Reply:
@yao-cqc I have a made a change to transform U(4) to SU(4) matrix. Please see in the PR